Update groups docs

Lior539 opened the pull request:
## Changes

* Updated docs to make it clearer what groups are and how to use them. The concept can quickly get confusing, so I'm looking for feedback on whether it's clear or not (especially the introduction)
* Renamed "Group analytics" to "Groups" in the docs. This because when searching seaching for "groups" using cmd+k, only the api docs show up (since "group analytics" doesnt match "groups"). It also aligns better with the app to call it "groups".
          - I considered renaming it in more places, but group analytics has SEO value, so I didn't rename it everywhere
 * also unified the groups docs in "getting started" and "product-analytics" to use the same component, so that we dont have 2 different docs for the same feature

ivanagas commented on contents/docs/_snippets/groups.mdx:
```diff
@@ -0,0 +1,327 @@
+
+1. For B2B SaaS apps, you can create a *company* group type to aggregate events at a company-level. This enables you to calculate metrics such as `number of daily active companies`, `company churn rate`, or `how many companies have adopted a new feature`.
+
+2. For a communication app like Slack, grouping events at a channel-level enables you to measure metrics like `average number of messages per channel`, `number of monthly active channels`, or `total number of channel participants`.
+
+3. For a social media app like Twitter, events can be grouped at a tweet-level – enabling you to measure `average number of replies per tweet`, or `total count of unique tweeters per month`.
```
```suggestion
2. For a collaborative, project-based app like Notion, Jira, or Figma, you can create a **project** group type. This enables you to calculate metrics like `project pageviews`, `users per project`, and `project engagement`.

3. For a communication app like Slack, you can create a **channel** group type. This enables you to measure metrics like `average number of messages per channel`, `number of monthly active channels`, or `total number of channel participants`.

4. For a social media app like Twitter, you can create a **post** group type. This enables you to measure `average number of replies per post`, or `total count of unique posters per month`.
```

ivanagas replied:
I think it's easier to follow if you do a consistent format.

ivanagas commented on contents/docs/_snippets/groups.mdx:
```diff
@@ -0,0 +1,327 @@
+
+> Groups are often confused with [cohorts](/docs/data/cohorts), but they each serve different purposes:
+>
+> * Groups aggregate events, and do not necessarily have to be connected to a user.
```
```suggestion
> - Groups aggregate events, and do not necessarily have to be connected to a user.
```

ivanagas commented on contents/docs/_snippets/groups.mdx:
```diff
@@ -0,0 +1,327 @@
+> Groups are often confused with [cohorts](/docs/data/cohorts), but they each serve different purposes:
+>
+> * Groups aggregate events, and do not necessarily have to be connected to a user.
+> * Cohorts represent a specific set of users – e.g., a list of users that all belong to the same company.
```
```suggestion
> - Cohorts represent a specific set of users – e.g., a list of users that all belong to the same company.
```

ivanagas commented on contents/docs/_snippets/groups.mdx:
```diff
@@ -0,0 +1,327 @@
+> * Groups aggregate events, and do not necessarily have to be connected to a user.
+> * Cohorts represent a specific set of users – e.g., a list of users that all belong to the same company.
+> 
+> Cohorts are easier to use and quicker to get started, whereas groups are more powerful. If your only goal is to create a list of **users** with something in common, we recommond cohorts instead of groups..
```
```suggestion
> Groups require instrumentation in your app to set up, while cohorts are created in PostHog and don't require additional instrumentation. This makes cohorts easier to use and quicker to get started. If your only goal is to create a list of **users** with something in common, we recommend cohorts instead of groups.
```

Lior539 replied:
Nice. I like how you've phrased it.

 Im not a big fan of the term "instrumentation", though, as it sounds vauge and complex

ivanagas commented on contents/docs/_snippets/groups.mdx:
```diff
@@ -0,0 +1,327 @@
+
+Groups are created and defined in your code when sending events. 
+
+In the example below, we create a group type `company`. Then, for each company, we set the `group key` as the unique identifier for that specifc company. This can be anything that helps you identify it, such as ID or domain.
```
```suggestion
In the example below, we create a group type `company`. Then, for each company, we set the group key as the unique identifier for that specific company. This can be anything that helps you identify it, such as ID, name, or domain.
```

Lior539 replied:
Name is not recommended, as mentioned in the call out below

ivanagas commented on contents/docs/_snippets/groups.mdx:
```diff
@@ -0,0 +1,327 @@
+
+We now have one `company`-type group with a key `company_id_in_your_db`. When we send the event `user_signed_up`, it will be attached to this newly created group.
+
+Note that a single event can only belong to a single group per group type. To elaborate on what we mean, we'll continue using the example above:
```
```suggestion
A single event can only belong to a single group per group type. To elaborate what we mean, we'll continue using the example above:
```

ivanagas commented on contents/docs/_snippets/groups.mdx:
```diff
@@ -0,0 +1,327 @@
+Note that a single event can only belong to a single group per group type. To elaborate on what we mean, we'll continue using the example above:
+
+* It is **NOT** possible to assign the a single `user_signed_up` event to two companies at the same time.
+* It is possible to assign this event to a different group type, for example `channel` or `tweet`, in addition to the `company` group type.
```
```suggestion
* It is possible to assign this event to a different group type, for example `channel` or `post`, in addition to the `company` group type.
```

ivanagas commented on contents/docs/_snippets/groups.mdx:
````diff
@@ -0,0 +1,327 @@
+```
+</MultiLanguage>
+
+
+
````
```suggestion
```

ivanagas commented on contents/docs/_snippets/groups.mdx:
```diff
@@ -0,0 +1,327 @@
+
+Continuing with the previous example of using `company` as our group type, we'll add `company_name`, `date_joined`, and `user_count` as additional properties.
+
+> **Note:** You must include at least one group property for a group to be visible in the "Persons & Groups" tab
```
```suggestion
> **Note:** You must include at least one group property for a group to be visible in the "Persons & Groups" tab.
```

ivanagas commented on contents/docs/_snippets/groups.mdx:
```diff
@@ -0,0 +1,327 @@
+
+</MultiLanguage>
+
+Properties on groups behave in the exact same way as properties on [persons](/docs/data/persons). They can also be used within experiments and feature flags to rollout features to specific groups.
```
```suggestion
Properties on groups behave in the same way as properties on [persons](/docs/data/persons). They can also be used within experiments and feature flags to rollout features to specific groups.
```

ivanagas commented on contents/docs/_snippets/groups.mdx:
```diff
@@ -0,0 +1,327 @@
+
+To view groups and their properties, head to the "Persons and Groups" tab on the navigation bar.
+
+From here you can select the group type you are interested in and view the groups and properties (by clicking the chevrons on the left).
```
```suggestion
From here, you can select the group type you are interested in and view the groups and properties (by clicking the chevrons on the left).
```

ivanagas commented on contents/docs/_snippets/groups.mdx:
```diff
@@ -0,0 +1,327 @@
+
+![View related people and groups](../../images/docs/user-guides/funnels/related-people-and-groups.png)
+
+### How to view group insights
```
```suggestion
### How to view group insights

```

ivanagas commented on contents/docs/_snippets/groups.mdx:
```diff
@@ -0,0 +1,327 @@
+
+For example, let's say that we wanted to see a graph showing how many different organizations have signed up recently:
+
+To do this, expand the menu next to the event you've chosen. This will a list of all the group types you have available. Then, by selecting "Unique organizations", this will show us a graph with the total number of organizations that have signed up (as opposed to individual users).
```
```suggestion
To do this, expand the menu next to the event you've chosen. This lists of all the group types available. Next, select the option for "Unique" with your group type name, such as "Unique organizations." This shows a graph with the total number of groups (organizations) that have signed up (as opposed to individual users).
```

ivanagas commented on contents/docs/_snippets/groups.mdx:
```diff
@@ -0,0 +1,327 @@
+
+Another place where group analytics can be used is within funnels.
+
+For example, you may want to understand how an organization moves from their first visit to eventually signing up. Yo can do this by setting the 'Aggregating by' field to 'Unique organizations'.
```
```suggestion
For example, you may want to understand how organizations move from their first visit to eventually signing up. You can do this by setting the "Aggregating by" field to "Unique organizations."
```

ivanagas commented on contents/docs/_snippets/groups.mdx:
```diff
@@ -0,0 +1,327 @@
+
+For example, you may want to understand how an organization moves from their first visit to eventually signing up. Yo can do this by setting the 'Aggregating by' field to 'Unique organizations'.
+
+This will show how many organizations have made it through, as well as the percentage of organizations that dropped off. It's also possible to see exactly which specifc groups dropped off at each step.
```
```suggestion
This shows how many organizations have made it through, as well as the percentage of organizations that dropped off. It's also possible to see exactly which specific groups dropped off at each step.
```

ivanagas commented on contents/docs/_snippets/groups.mdx:
```diff
@@ -0,0 +1,327 @@
+
+Groups in feature flags enable you to rollout a feature by group type (like `company`), instead of users.
+
+To do this, create a feature flag as you normally would, except you need to select the group type you wish to "Match by", using the drop down in the "Release conditions" section:
```
```suggestion
To do this, create a feature flag as you normally would, and select the group type you wish to "Match by", using the drop down in the "Release conditions" section:
```

ivanagas commented on contents/docs/_snippets/groups.mdx:
```diff
@@ -0,0 +1,327 @@
+
+![Groups in project settings](../../images/docs/user-guides/funnels/groups-in-feature-flags.mp4)
+
+You will also need to update your event tracking code in order for the feature flag to be able to determine the groups of the current user.
```
```suggestion
You also need to update your event tracking code for the feature flag to determine the groups of the current user.
```

ivanagas commented on contents/docs/_snippets/groups.mdx:
```diff
@@ -0,0 +1,327 @@
+
+### Renaming group types
+
+You can change how group types are displayed in your Insights interface and throughout PostHog by using the Project settings.
```
```suggestion
You can change how group types are displayed in the insights interface and throughout PostHog by in project settings.
```

ivanagas commented on contents/docs/_snippets/groups.mdx:
```diff
@@ -0,0 +1,327 @@
+-   Multiple groups of the same type cannot assigned to a single event (e.g., Company A & Company B).
+-   Groups are not currently supported for the following insights:
+    -   Lifecycle - Expected soon.
+    -   User Paths - These only support user level analytics.
```
```suggestion
    -   User paths - These only support user level analytics.
```

ivanagas commented on contents/docs/_snippets/groups.mdx:
```diff
@@ -0,0 +1,327 @@
+    -   Lifecycle - Expected soon.
+    -   User Paths - These only support user level analytics.
+-   Only groups with known properties are shown under 'Persons & groups'.
+-   Currently there is no functionality within the app to delete groups. If you'd like to delete a group, please [contact customer support](/growth/customer-support#we-aim-to-delight).
```
```suggestion
-   Currently there is no functionality within the app to delete groups. If you'd need to delete a group for privacy or security reasons, please [use the support modal](https://app.posthog.com/home#supportModal=support%3Adata_management).
```

Lior539 replied:
```suggestion
-   Currently there is no functionality within the app to delete groups. If you need to delete a group for privacy or security reasons, please [use the support modal](https://app.posthog.com/home#supportModal=support%3Adata_management).```

ivanagas commented on contents/docs/getting-started/group-analytics.mdx:
```diff
@@ -1,70 +1,9 @@
-import CreateGroupType from "./\_snippets/group-type-create.mdx"
-import ChatGroupType from "./\_snippets/group-type-create-chat.mdx"
-import SetGroupProperties from "./\_snippets/set-group-properties.mdx"
+import JSGroups from "../\_snippets/groups.mdx"
```
We shouldn't have two pages with the exact same content if it is this long. We can just link to the `product-analytics/group-analytics` page in the getting started sidebar if we don't want another page.

ivanagas replied:
Alternatively, the getting started section should have some basic details on set up and usage while the main page has the full context. You can use a part as a snippet in that situation.

Lior539 replied:
Fair point. 

> Alternatively, the getting started section should have some basic details on set up and usage while the main page has the full context. You can use a part as a snippet in that situation.

I think I'll go with this solution

andyvan-ph replied:
Another issue with this snippet approach is the page nav in the right sidebar doesn't load. 

<img width="325" alt="Screenshot 2023-05-23 at 10 39 59" src="https://github.com/PostHog/posthog.com/assets/92976667/6c0f6fc6-95b2-4b0a-bd7c-3dcf1e9c1c22">

ivanagas commented on contents/docs/getting-started/group-analytics.mdx:
```diff
@@ -1,70 +1,9 @@
 ---
-title: Group analytics
```
I feel strongly that we shouldn't rename this. Groups is a very generic term, group analytics explains it a lot better. We mention group analytics everywhere already, and importantly, on the pricing page. It's also better for SEO. 

Groups could also refer to user groups or any other type of group.

ivanagas replied:
Also for the Algolia search, groups vs group analytics, I'm pretty sure we can set up an override that lets us show group analytics for the term "groups". Do you know @smallbrownbike ?

Lior539 replied:
I think if we can override the algolia search, then thats a good compromise. My main concern is that we refer to it as "Groups" in the app, so people are more likely to search for "groups" and not "group analytics"

@corywatilo maybe you know how to override?

ivanagas commented on contents/docs/product-analytics/group-analytics.mdx:
```diff
@@ -1,5 +1,5 @@
 ---
-title: Group analytics
```
Don't rename.

ivanagas commented on contents/docs/product-analytics/group-analytics.mdx:
```diff
@@ -10,282 +10,6 @@ related:
 ---
 
-> **Note: ** This is a paid feature and is not available on the open-source or free cloud plan. Learn more [here](/pricing).
+import JSGroups from "../\_snippets/groups.mdx"
```
If we are just using this snippet on this page, it shouldn't be a snippet.

ivanagas commented on src/sidebars/docs.json:
```diff
@@ -35,7 +35,7 @@
             },
             {
-                "name": "Group analytics",
+                "name": "Groups",
```
Don't rename

ivanagas commented on src/sidebars/docs.json:
```diff
@@ -35,7 +35,7 @@
             {
-                "name": "Group analytics",
+                "name": "Groups",
                 "url": "/docs/getting-started/group-analytics"
```
We could link to /docs/product-analytics/group-analytics here

ivanagas reviewed (changes requested):
Some changes:
- I feel strongly that we should rename group analytics to groups
- We should have the same long piece of content on two pages, small snippets are fine though.
- Either link to the main product-analytics/group-analytics page or create a new getting started page.

Lior539 commented on contents/docs/_snippets/groups.mdx:
```diff
@@ -0,0 +1,327 @@
+> **Note: ** This is a paid feature and is not available on the open source or free cloud plan. See our [pricing page](/pricing) for more.
+
+**Groups** are a powerful feature in PostHog that aggregate events based on entities. They enable you to track trends, insights, and dashboards at an entity-level – as opposed to regular events, which only show you insights on a user-level.
```
```suggestion
**Groups** are a powerful feature in PostHog that aggregate events based on entities, such as organizations or companies. They enable you to analyze trends, insights, and dashboards at an entity-level, as opposed to a user-level.
```

Lior539 replied:
```suggestion
**Groups** are a powerful feature in PostHog that aggregate events based on entities, such as organizations or companies. They enable you to analyze trends, insights, and dashboards at an entity-level, as opposed to a user-level.```

Lior539 commented on contents/docs/_snippets/groups.mdx:
```diff
@@ -0,0 +1,327 @@
+
+To clarify what we mean, let's look at a few examples:
+
+1. For B2B SaaS apps, you can create a *company* group type to aggregate events at a company-level. This enables you to calculate metrics such as `number of daily active companies`, `company churn rate`, or `how many companies have adopted a new feature`.
```
```suggestion
1. For B2B SaaS apps, you can create a **company** group type. This enables you to aggregate events at a company-level, and calculate metrics such as `number of daily active companies`, `company churn rate`, or `how many companies have adopted a new feature`.```

Lior539 commented on contents/docs/_snippets/groups.mdx:
```diff
@@ -0,0 +1,327 @@
+
+1. For B2B SaaS apps, you can create a *company* group type to aggregate events at a company-level. This enables you to calculate metrics such as `number of daily active companies`, `company churn rate`, or `how many companies have adopted a new feature`.
+
+2. For a communication app like Slack, grouping events at a channel-level enables you to measure metrics like `average number of messages per channel`, `number of monthly active channels`, or `total number of channel participants`.
+
+3. For a social media app like Twitter, events can be grouped at a tweet-level – enabling you to measure `average number of replies per tweet`, or `total count of unique tweeters per month`.
```
```suggestion
2. For collaborative, project-based apps like Notion, Jira, or Figma, you can create a **project** group type. This enables you to calculate metrics like `project pageviews`, `users per project`, and `project engagement`.

3. For a communication app like Slack, you can create a **channel** group type. This enables you to measure metrics like `average number of messages per channel`, `number of monthly active channels`, or `total number of channel participants`.

4. For a social media app like Twitter, you can create a **post** group type. This enables you to measure `average number of replies per post`, or `total count of unique posters per month`.
```

Lior539 commented on contents/docs/_snippets/groups.mdx:
```diff
@@ -0,0 +1,327 @@
+
+Groups are created and defined in your code when sending events. 
+
+In the example below, we create a group type `company`. Then, for each company, we set the `group key` as the unique identifier for that specifc company. This can be anything that helps you identify it, such as ID or domain.
```
```suggestion
In the example below, we create a group type `company`. Then, for each company, we set the `group key` as the unique identifier for that specific company. This can be anything that helps you identify it, such as ID or domain.
```

andyvan-ph commented on contents/docs/_snippets/groups.mdx:
````diff
@@ -0,0 +1,328 @@
+
+4. For a social media app like Twitter, you can create a **post** group type. This enables you to measure `average number of replies per post`, or `total count of unique posters per month`.
+
+> Groups are often confused with [cohorts](/docs/data/cohorts), but they each serve different purposes:
+>
+> - Groups aggregate events, and do not necessarily have to be connected to a user.
+> - Cohorts represent a specific set of users – e.g., a list of users that all belong to the same company.
+> 
+> Groups require additional code in your app to set up, while cohorts are created in PostHog and don't require additional code. This makes cohorts easier to use and quicker to get started. If your only goal is to create a list of **users** with something in common, we recommend cohorts instead of groups.```
````
```suggestion
## The difference between groups and cohorts

Groups are often confused with [cohorts](/docs/data/cohorts), but they each serve different purposes:

- Groups aggregate events, and do not necessarily have to be connected to a user.
- Cohorts represent a specific set of users – e.g., a list of users that all belong to the same company.

Groups require additional code in your app to set up, while cohorts are created in PostHog and don't require additional code. This makes cohorts easier to use and quicker to get started. If your only goal is to create a list of **users** with something in common, we recommend cohorts instead of groups.```
```

andyvan-ph replied:
Better have this with a heading than a boxout. It's quite important and makes the section more searchable.

Lior539 commented on contents/docs/_snippets/groups.mdx:
````diff
@@ -0,0 +1,328 @@
+-   Only groups with known properties are shown under 'Persons & groups'.
+-   Currently there is no functionality within the app to delete groups. If you need to delete a group for privacy or security reasons, please [use the support modal](https://app.posthog.com/home#supportModal=support%3Adata_management).```
+
+## Further reading
+
+- [Frontend vs Backend Group Analytics](/tutorials/frontend-vs-backend-group-analytics).
+- [When and how to run group-targeted A/B tests](/blog/running-group-targeted-ab-tests)
````
Re the tutorial: Fair enough, will remove.
Re the ab tests blog: I feel quite strongly that we should link to it, as it's a relevant topic for this doc. Perhaps it should have been a tutorial though

Lior539 commented:
* I've renamed Groups back to Group Analytics
* made the changes re differentiating getting started and product analytics page
* split the snippet into multiple snippets (so that the right nav bar doesnt break)

andyvan-ph commented:
The more I look at this, the more I wonder if Group analytics should even be in Getting started at all. It's confusing to have two pages on the same topic, and makes maintenance more difficult. The component solution was a good idea, but you still end up with two pages with the same name / content, but different URLs. It's also, afaik, the only paid-for feature that is in Getting started.

andyvan-ph commented on contents/docs/_snippets/groups-intro.mdx:
```diff
@@ -0,0 +1,11 @@
+**Groups** are a powerful feature in PostHog that aggregate events based on entities, such as organizations or companies. They enable you to analyze trends, insights, and dashboards at an entity-level, as opposed to a user-level.
```
```suggestion
## What is group analytics?

**Groups** are a powerful feature in PostHog that aggregate events based on entities, such as organizations or companies. They enable you to analyze trends, insights, and dashboards at an entity-level, as opposed to a user-level.
```

andyvan-ph commented on contents/docs/getting-started/group-analytics.mdx:
```diff
@@ -4,75 +4,34 @@ nextPage: ./next-steps.mdx
 
-Group analytics is a powerful feature in PostHog that allows you to perform analytics on entities (e.g. companies, teams etc.) other than single users.
-This feature is slightly more advanced, but opens up entirely new avenues for analytics that are essential for most B2B use cases.
+<JSGroupsIntro />
```
```suggestion
## What are groups in PostHog?

<JSGroupsIntro />
```

Lior539 commented:
@andyvan-ph yeah I think you're right. Groups really isnt a "getting started" kind of feature. I think it's more advanced usage. I'll remove it

andyvan-ph commented:
@Lior539 Let's slow down a bit before we make that call. Let's table this for the next sprint meeting. There was probably a reason if was there, so worth checking in before we make a big change. I'm Cory and James atm – will get their take.

ivanagas commented:
I think (and had some light agreement in the sprint meeting) that we should move the getting started group analytics content into send events (maybe a bit shorter), and link to the group analytics under product analytics.

andyvan-ph commented:
> I think (and had some light agreement in the sprint meeting) that we should move the getting started group analytics content into send events (maybe a bit shorter), and link to the group analytics under product analytics.

This sounds sensible to me.

Lior539 commented:
I think that makes sense. I would prefer to do rework the Getting Started altogether, and rethink what to show there (e.g., we don't mention session replays at all). Can we merge this for now, and I'll make the other updates in a separate PR?

andyvan-ph commented:
> I think that makes sense. I would prefer to do rework the Getting Started altogether, and rethink what to show there (e.g., we don't mention session replays at all). Can we merge this for now, and I'll make the other updates in a separate PR?

Yup, that makes sense to me. I agree we could probably improve the the Getting started section a fair bit. If you're planning major changes, it's worth outlining your thoughts in an issue beforehand.

andyvan-ph reviewed (approved):
There's a conflict, but good to go once that's resolved.

andyvan-ph reviewed (approved):
There's a conflict, but good to go once that's resolved.
